compiler/ast/ast_types.nim

```diff
@@ -1615,6 +1615,20 @@ type
     name*: Rope
     path*: PNode              ## can be a string literal!
 
+  LibId* = object
+    ## Identifies a ``TLib`` instance. The default value means 'none'.
```

Comment:
Might be worth adding a `unknownLibId` or the like as a constant below to further 'reserve' the concept of the default value `LibId(module: 0, index: 0)` as none.

---

Update: reading more of the PR, there is a similar tuple type for IC, so we wouldn't end up using that const/this object type all that much so it might have less mileage in that regard. Seems less worth it practically, mostly there to signal intent.